aptos-move/move-examples/marketplace/sources/coin_listing.move

```diff
@@ -0,0 +1,1156 @@
+        object: Object<Listing>,
+    ) acquires FixedPriceListing {
+        let expected_seller_addr = signer::address_of(seller);
+        let (actual_seller_addr, _fee_schedule) = listing::close(object, expected_seller_addr);
+        assert!(expected_seller_addr == actual_seller_addr, error::permission_denied(ENOT_SELLER));
+
+        let listing_addr = object::object_address(&object);
+        assert!(exists<FixedPriceListing<CoinType>>(listing_addr), error::not_found(ENO_LISTING));
+        let FixedPriceListing {
+            price: _,
+            purchase_event,
+        } = move_from<FixedPriceListing<CoinType>>(listing_addr);
+
+        let purchase_event_data = PurchaseEvent {
+            purchaser: expected_seller_addr,
+            price: 0,
+            commission: 0,
+            royalties: 0,
+        };
+        event::emit_event(&mut purchase_event, purchase_event_data);
+        event::destroy_handle(purchase_event);
+    }
+
+    /// Make a bid on a listing. If the listing comes in near the end of an auction, the auction
+    /// may be extended to give at least minimum_bid_time_before_end time remaining in the auction.
+    public entry fun bid<CoinType>(
+        bidder: &signer,
+        object: Object<Listing>,
+        bid_amount: u64,
+    ) acquires AuctionListing {
+        let listing_addr = listing::assert_started(&object);
+        assert!(exists<AuctionListing<CoinType>>(listing_addr), error::not_found(ENO_LISTING));
+        let auction_listing = borrow_global_mut<AuctionListing<CoinType>>(listing_addr);
+
+        let now = timestamp::now_seconds();
+        assert!(now < auction_listing.auction_end_time, error::invalid_state(EAUCTION_ENDED));
+
+        let (previous_bidder, previous_bid, minimum_bid) = if (option::is_some(&auction_listing.current_bid)) {
+            let Bid { bidder, coins } = option::extract(&mut auction_listing.current_bid);
+            let current_bid = coin::value(&coins);
+            coin::deposit(bidder, coins);
+            (option::some(bidder), option::some(current_bid), current_bid + auction_listing.bid_increment)
+        } else {
+            (option::none(), option::none(), auction_listing.starting_bid)
+        };
+
+        assert!(bid_amount >= minimum_bid, error::invalid_argument(EBID_TOO_LOW));
+        let coins = coin::withdraw<CoinType>(bidder, bid_amount);
+        let bid = Bid {
+            bidder: signer::address_of(bidder),
+            coins,
+        };
+        option::fill(&mut auction_listing.current_bid, bid);
+
+        let fee_schedule = listing::fee_schedule(object);
+        coin::transfer<CoinType>(
+            bidder,
+            fee_schedule::fee_address(fee_schedule),
+            fee_schedule::bidding_fee(fee_schedule, bid_amount),
+        );
+
+        let now = timestamp::now_seconds();
+        let current_end_time = auction_listing.auction_end_time;
+        let minimum_end_time = now + auction_listing.minimum_bid_time_before_end;
```

Comment:
if the seller cannot cancel the auction and the bid_increment is too low, would this allow ppl to endlessly extend the auction duration?
Why do we need this?

Reply:
this is actually quite common... we can update based upon community feedback